sim/core/buffs.go

```diff
@@ -1524,3 +1529,49 @@ func FocusMagicAura(caster *Unit, target *Unit) (*Aura, *Aura) {
 
 	return casterAura, aura
 }
+
+// Builds an ExclusiveEffect representing a SpellHaste bonus multiplier
+// spellHastePercent should be given as the percent value i.E. 0.05 for +5%
+func SpellHasteBonusEffect(aura *Aura, spellHastePercent float64) *ExclusiveEffect {
+	return aura.NewExclusiveEffect("SpellHasteBonus", false, ExclusiveEffect{
+		Priority: spellHastePercent,
+		OnGain: func(ee *ExclusiveEffect, sim *Simulation) {
+			ee.Aura.Unit.PseudoStats.CastSpeedMultiplier *= (1 + ee.Priority)
+			ee.Aura.Unit.updateCastSpeed()
+		},
+		OnExpire: func(ee *ExclusiveEffect, sim *Simulation) {
+			ee.Aura.Unit.PseudoStats.CastSpeedMultiplier /= (1 + ee.Priority)
+			ee.Aura.Unit.updateCastSpeed()
+		},
+	})
+}
+
+func WrathOfAirAura(character *Character) *Aura {
+	aura := character.GetOrRegisterAura(Aura{
+		Label:      "Wrath of Air",
+		ActionID:   ActionID{SpellID: 3738},
+		Duration:   NeverExpires,
+		BuildPhase: CharacterBuildPhaseBuffs,
+		OnReset: func(aura *Aura, sim *Simulation) {
+			aura.Activate(sim)
+		},
+	})
+
+	SpellHasteBonusEffect(aura, 0.05)
+	return aura
+}
+
+func MindQuickeningAura(character *Character) *Aura {
+	aura := character.GetOrRegisterAura(Aura{
+		Label:      "Mind Quickening",
+		ActionID:   ActionID{SpellID: 49868},
+		Duration:   NeverExpires,
+		BuildPhase: CharacterBuildPhaseBuffs,
+		OnReset: func(aura *Aura, sim *Simulation) {
+			aura.Activate(sim)
+		},
+	})
+
+	SpellHasteBonusEffect(aura, 0.05)
+	return aura
+}
```

Comment:
Would you mind also adding Moonkin aura here? I saw you included it in the UI so we should include it in the exclusive effects

Reply:
Should be available now. Removed it from the presets where possible and another haste buff was also present. Would love to add the other raid buffs as well but that will happen on another branch.
Gotta see how I can do that without running into many merge conflicts with all the other branches like warrior or shaman also implementing some buffs already 😁